### ISSUE TITLE
Products index's setting get removed over night

### ISSUE THREAD

**Issue by greatwitenorth** [algolia/algoliasearch-magento#690]
I'm noticing that when I login to Algolia's explorer and view my product index, that each morning all the settings for the index have been removed. All the faceting abilities are gone, along with ranking criteria. The 'Advice' tab is showing that there's some outstanding issues that need addressing (since all the settings have been wiped out).

I then go into my magento admin and visit the configuration screen for Algolia, and click save. I visit the Algolia explorer again, and now all the product index's settings have been restored.

I see that during batch operations, that a _tmp product indexed is made, then moved. Is it possible that when this tmp index is created that all the settings are not created as well? Then once the 'move' operation is done, then it effectively overwrites the settings on the main product index?

I'm using:
Algolia plugin v1.7.2
Magento v1.9.0.1

**Comment by JanPetr** [comment]
Hello @greatwitenorth,

thanks for the report. It's possible but I'm currently not able to reproduce and fix. If you'll manage to find a concrete use case when it happens, please let me know.
In the meantime I will definitely investigate and find the root cause.

Thank you.

Best,

**Comment by greatwitenorth** [comment]
I'm not sure if you have access to my algolia account, but you can see the above scenario there right now. All the settings for the Index have been wiped out. If I got to my magento admin and visit the algolia settings and click save, then all the settings come back. 

I can give you my algolia application ID if that helps, is it better to private message?

**Comment by greatwitenorth** [comment]
Any word on this? I'm still seeing this behavior.

**Comment by JanPetr** [comment]
Hello @greatwitenorth,
sorry for not getting back to you, I wasn't available for a while.

Did the issue still persist? Did you manage to find a reliable way on how to reproduce the issue or is it appearing randomly?
I went through the whole code base and couldn't find a way where it wipes the settings out.
Could you send your app ID to support+magento@algolia.com? It'd really help me.

Thanks!

**Comment by JanPetr** [comment]
Solved via Algolia support. The issue was in two concurrent crons, which launched a queue runner indexer at the very same time. So the queue runner got launched twice and performed two moves of the TMP Index and the second move wiped out the settings.

We implemented a DB transaction (#713, #715) into a queue runner so the issue should not repeat in a new versions. Thanks for your cooperation @greatwitenorth 💯
